internal/stack/stacks.go

```diff
@@ -136,19 +148,43 @@ func (p *stackParser) parseStack(line string) (Stack, error) {
 		fullStack.WriteString(line)
 		fullStack.WriteByte('\n') // scanner trims the newline
 
-		// The first line after the header is the top of the stack.
+		if len(line) == 0 {
+			continue
+		}
+
+		funcName, err := parseFuncName(line)
+		if err != nil {
+			return Stack{}, fmt.Errorf("parse function: %w", err)
+		}
+		funcs[funcName] = struct{}{}
 		if firstFunction == "" {
-			firstFunction, err = parseFirstFunc(line)
-			if err != nil {
-				return Stack{}, fmt.Errorf("extract function: %w", err)
+			firstFunction = funcName
+		}
+
+		// The function name is usually followed by a line in the form:
```

Comment:
not exactly sure what usually means here -- are there cases where it isn't followed by the file:line that we can enumerate? or is this trying to be defensive to unknown cases?

Reply:
It's just being defensive. I'll clarify.